File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Load Cell Bricklet

    ' Callback subroutine for weight reached callback (parameter has unit g)
    Sub WeightReachedCB(ByVal sender As BrickletLoadCell, ByVal weight As Integer)
        Console.WriteLine("Weight: " + weight.ToString() + " g")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim lc As New BrickletLoadCell(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 1 second (1000ms)
        lc.SetDebouncePeriod(1000)

        ' Register weight reached callback to subroutine WeightReachedCB
        AddHandler lc.WeightReached, AddressOf WeightReachedCB

        ' Configure threshold for weight "greater than 200 g" (unit is g)
        lc.SetWeightCallbackThreshold(">"C, 200, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleThreshold.pas
program ExampleThreshold;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletLoadCell;

type
  TExample = class
  private
    ipcon: TIPConnection;
    lc: TBrickletLoadCell;
  public
    procedure WeightReachedCB(sender: TBrickletLoadCell; const weight: longint);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change to your UID }

var
  e: TExample;

{ Callback procedure for weight greater than 2 kg (parameter has unit g) }
procedure TExample.WeightReachedCB(sender: TBrickletLoadCell; const weight: longint);
begin
  WriteLn(Format('Weight: %f kg', [weight/1000.0]));
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  lc := TBrickletLoadCell.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get threshold callbacks with a debounce time of 1 second (1000ms) }
  lc.SetDebouncePeriod(1000);

  { Register threshold reached callback to procedure WeightReachedCB }
  lc.OnWeightReached := {$ifdef FPC}@{$endif}WeightReachedCB;

  { Configure threshold for "greater than 2 kg" (unit is g) }
  lc.SetWeightCallbackThreshold('>', 2*1000, 0);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
